restli-tools/src/main/java/com/linkedin/restli/tools/data/ExtensionSchemaValidationCmdLineApp.java

```diff
@@ -134,10 +155,75 @@ private static void parseAndValidateExtensionSchemas(String resolverPath, File i
           .stream()
           .filter(f -> !((RecordDataSchema) topLevelDataSchema).isFieldFromIncludes(f))
           .collect(Collectors.toList());
-      // TODO : Call annotation process to validate each field's annotation, need to create schemaVisitor and handler for extension annotation.
+
+      checkExtensionSchemaFields(extensionSchemaFields);
+    }
+  }
+
+  private static void checkExtensionSchemaFields(List<RecordDataSchema.Field> extensionSchemaFields)
+  {
+    for (RecordDataSchema.Field field : extensionSchemaFields)
+    {
+      // check extension schema field annotation
+      Map<String, Object> properties = field.getProperties();
+      if (properties.isEmpty() || properties.keySet().size() != 1 || !properties.containsKey(EXTENSION_ANNOTATION_NAMESPACE))
+      {
+        _logger.error("The field [{}] of extension schema must and only be annotated with 'extension'", field.getName());
+        System.exit(1);
+      }
+      Object dataElement = properties.get(EXTENSION_ANNOTATION_NAMESPACE);
+
+      ValidationOptions validationOptions =
+          new ValidationOptions(RequiredMode.MUST_BE_PRESENT, CoercionMode.STRING_TO_PRIMITIVE, UnrecognizedFieldMode.DISALLOW);
+      try
+      {
+        if (!(dataElement instanceof DataMap))
+        {
+          _logger.error("extension schema annotation is not a datamap!");
+          System.exit(1);
+        }
+        DataSchema extensionSchemaAnnotationSchema = new ExtensionSchemaAnnotation().schema();
+        ValidationResult result = ValidateDataAgainstSchema.validate(dataElement, extensionSchemaAnnotationSchema, validationOptions);
+        if (!result.isValid())
+        {
+          _logger.error("extension schema annotation is not valid: " + result.getMessages());
+          System.exit(1);
+        }
+      }
+      catch (Exception e)
+      {
+        _logger.error("Error while checking extension schema field annotation: " + e.getMessage());
+        System.exit(1);
+      }
+      checkExtensionSchemaFieldSchema(field.getType());
```

Comment:
Seems this line can be combined into the try catch clause. I personally would try to minimize the number of exit path (the statement System.exit(1))

Reply:
Sorry, I don't quite understand it, can you please explain more?